aws/resource_aws_pinpoint_app_test.go

```diff
@@ -0,0 +1,249 @@
+				),
+			},
+		},
+	})
+}
+
+func TestAccAWSPinpointApp_CampaignHookLambda(t *testing.T) {
+	attributes := make(map[string]string)
+	resourceName := "aws_pinpoint_app.test_app"
+	appName := "terraform-test-pinpointapp-campaignhooklambda"
+	lambdaName := "test-pinpoint-lambda"
+	resource.Test(t, resource.TestCase{
+		PreCheck:      func() { testAccPreCheck(t) },
+		IDRefreshName: "aws_pinpoint_app.test_app",
+		Providers:     testAccProviders,
+		CheckDestroy:  testAccCheckAWSPinpointAppDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSPinpointAppConfig_CampaignHookLambda(appName, lambdaName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSPinpointAppExists(resourceName, attributes),
+					resource.TestCheckResourceAttr(resourceName, "campaign_hook.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "campaign_hook.4129384237.mode", "DELIVERY"),
+					resource.TestCheckResourceAttr(resourceName, "limits.#", "0"),
+					resource.TestCheckResourceAttr(resourceName, "quiet_time.#", "0"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAWSPinpointApp_Limits(t *testing.T) {
+	attributes := make(map[string]string)
+	resourceName := "aws_pinpoint_app.test_app"
+	resource.Test(t, resource.TestCase{
+		PreCheck:      func() { testAccPreCheck(t) },
+		IDRefreshName: "aws_pinpoint_app.test_app",
+		Providers:     testAccProviders,
+		CheckDestroy:  testAccCheckAWSPinpointAppDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSPinpointAppConfig_Limits,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSPinpointAppExists(resourceName, attributes),
+					resource.TestCheckResourceAttr(resourceName, "campaign_hook.#", "0"),
+					resource.TestCheckResourceAttr(resourceName, "limits.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "limits.1765820463.total", "500"),
+					resource.TestCheckResourceAttr(resourceName, "quiet_time.#", "0"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAWSPinpointApp_QuietTime(t *testing.T) {
+	attributes := make(map[string]string)
+	resourceName := "aws_pinpoint_app.test_app"
+	resource.Test(t, resource.TestCase{
+		PreCheck:      func() { testAccPreCheck(t) },
+		IDRefreshName: "aws_pinpoint_app.test_app",
+		Providers:     testAccProviders,
+		CheckDestroy:  testAccCheckAWSPinpointAppDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSPinpointAppConfig_QuietTime,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSPinpointAppExists(resourceName, attributes),
+					resource.TestCheckResourceAttr(resourceName, "campaign_hook.#", "0"),
+					resource.TestCheckResourceAttr(resourceName, "limits.#", "0"),
+					resource.TestCheckResourceAttr(resourceName, "quiet_time.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "quiet_time.537668748.start", "00:00"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckAWSPinpointAppExists(n string, attributes map[string]string) resource.TestCheckFunc {
```

Comment:
It appears `attributes` is unused in this function. We prefer to use the second parameter for `Exists` test functions to return the API object returned by the get/describe/read function of the SDK so we can use it in later acceptance test check functions if necessary. e.g.

```go
func TestAccAWSPinpointApp_basic(t *testing.T) {
  var application pinpoint.ApplicationResponse
// ...
  testAccCheckAWSPinpointAppExists(resourceName, &application),
```

And its definition:

```go
func testAccCheckAWSPinpointAppExists(n string, application *pinpoint.ApplicationResponse) resource.TestCheckFunc {
// ...
params := &pinpoint.GetAppInput{
  ApplicationId: aws.String(rs.Primary.ID),
}
output, err := conn.GetApp(params)

if err != nil {
  return err
}

*application = *output.ApplicationResponse

return nil
```

Reply:
Fixed in https://github.com/terraform-providers/terraform-provider-aws/pull/5956/commits/be7dc0047e5e3ec5d966f2a51c702546a27cfb95